crates/anvil/src/eth/backend/executor.rs

```diff
@@ -284,15 +286,49 @@ impl<'a, 'b, DB: Db + ?Sized, Validator: TransactionValidator> Iterator
         let nonce = account.nonce;
 
         // records all call and step traces
-        let mut inspector = Inspector::default().with_tracing();
-        if self.enable_steps_tracing {
-            inspector = inspector.with_steps_tracing();
+        let mut inspector = Inspector::default();
+        if !self.disable_tracing {
+            inspector = inspector.with_tracing();
+            if self.enable_steps_tracing {
+                inspector = inspector.with_steps_tracing();
+            }
         }
 
-        let exec_result = {
+        let exec_result = if !self.disable_tracing {
             let mut evm =
                 foundry_evm::utils::new_evm_with_inspector(&mut *self.db, env, &mut inspector);
 
+            trace!(target: "backend", "[{:?}] executing", transaction.hash());
+            // transact and commit the transaction
+            match evm.transact_commit() {
+                Ok(exec_result) => exec_result,
+                Err(err) => {
+                    warn!(target: "backend", "[{:?}] failed to execute: {:?}", transaction.hash(), err);
+                    match err {
+                        EVMError::Database(err) => {
+                            return Some(TransactionExecutionOutcome::DatabaseError(
+                                transaction,
+                                err,
```

Comment:
I think we don't need the duplication here if we make the if else block return the result?

Reply:
The issue was building the `evm` with different types, one with an inspector and one without.